checker/jdk/nullness/src/javax/xml/parsers/DocumentBuilderFactory.java

```diff
@@ -0,0 +1,613 @@
+     * <code>DocumentBuilderFactory</code>. This static method creates
+     * a new factory instance.
+     * This method uses the following ordered lookup procedure to determine
+     * the <code>DocumentBuilderFactory</code> implementation class to
+     * load:
+     * <ul>
+     * <li>
+     * Use the <code>javax.xml.parsers.DocumentBuilderFactory</code> system
+     * property.
+     * </li>
+     * <li>
+     * Use the properties file "lib/jaxp.properties" in the JRE directory.
+     * This configuration file is in standard <code>java.util.Properties
+     * </code> format and contains the fully qualified name of the
+     * implementation class with the key being the system property defined
+     * above.
+     *
+     * The jaxp.properties file is read only once by the JAXP implementation
+     * and it's values are then cached for future use.  If the file does not exist
+     * when the first attempt is made to read from it, no further attempts are
+     * made to check for its existence.  It is not possible to change the value
+     * of any property in jaxp.properties after it has been read for the first time.
+     * </li>
+     * <li>
+     * Uses the service-provider loading facilities, defined by the
+     * {@link java.util.ServiceLoader} class, to attempt to locate and load an
+     * implementation of the service using the {@linkplain
+     * java.util.ServiceLoader#load(java.lang.Class) default loading mechanism}:
+     * the service-provider loading facility will use the {@linkplain
+     * java.lang.Thread#getContextClassLoader() current thread's context class loader}
+     * to attempt to load the service. If the context class
+     * loader is null, the {@linkplain
+     * ClassLoader#getSystemClassLoader() system class loader} will be used.
+     * </li>
+     * <li>
+     * Otherwise, the system-default implementation is returned.
+     * </li>
+     * </ul>
+     *
+     * Once an application has obtained a reference to a
+     * <code>DocumentBuilderFactory</code> it can use the factory to
+     * configure and obtain parser instances.
+     *
+     *
+     * <h2>Tip for Trouble-shooting</h2>
+     * <p>Setting the <code>jaxp.debug</code> system property will cause
+     * this method to print a lot of debug messages
+     * to <code>System.err</code> about what it is doing and where it is looking at.</p>
+     *
+     * <p> If you have problems loading {@link DocumentBuilder}s, try:</p>
+     * <pre>
+     * java -Djaxp.debug=1 YourProgram ....
+     * </pre>
+     *
+     * @return New instance of a <code>DocumentBuilderFactory</code>
+     *
+     * @throws FactoryConfigurationError in case of {@linkplain
+     * java.util.ServiceConfigurationError service configuration error} or if
+     * the implementation is not available or cannot be instantiated.
+     */
+    @SuppressWarnings("nullness")    // find method in FactoryFinder accepts @Nullable args whereas here the requirement is @NonNull
```

Comment:
I don't understand this warning suppression.  What error is thrown if it is removed?

Reply:
No error will be thrown while running with annotated `FactoryFinder.java`. While annotating. my approach is to annotate files independent of others, keeping in mind the scenario of partially annotated library. 
Will change it.
-Thanks.